docs/commands.md

```diff
@@ -0,0 +1,153 @@
+# CLI Commands
+
+## `init`
+
+Creates a new Function project at _`path`_. If _`path`_ is unspecified, assumes the current directory. If _`path`_ does not exist, it will be created. The user can specify the runtime and trigger with flags.
+
+Similar `kn` command: none.
+
+```console
+faas init <path> [-l <runtime> -t <trigger>]
+```
+
+When run as a `kn` plugin.
+
+```console
+kn faas init <path> [-l <runtime> -t <trigger>]
+```
+
+## `build`
+
+Builds the Function project in the current directory. Reads the `.faas.yaml` file to determine image name and repository. If both of these values are unset in the configuration file, the user is prompted to provide a repository, from there an image name can be derived. The image name and repository may also be specified as flags, as can the path to the project.
+
+The value(s) provided for image and repository are persisted to the `.faas.yaml` file so that subsequent invocations do not require the user to specify these again.
+
+Similar `kn` command: none.
+
+```console
+faas build [-i <image> -r <repository> -p <path>]
+```
+
+When run as a `kn` plugin.
+
+```console
+kn faas build [-i <image> -r <repository> -p <path>]
+```
+
+## `run`
+
+Runs the Function project locally in the container. If a container has not yet been created, prompts the user to run `faas build`. Note: there is no option to specify a path to the project.
+
+Similar `kn` command: none.
+
+```console
+faas run
+```
+
+When run as a `kn` plugin.
+
+```console
+kn faas run
+```
+
+## `deploy`
+
+Deploys the Function project in the current directory. The user may specify a path to the project directory as a flag. Reads the `.faas.yaml` configuration file to determine the image name. Derives the service name from the project name. There is no command line option to specify the image name, although this can be changed in `.faas.yaml`. There is no mechanism by which the user can specify the service name. By default, the image is deployed to the namespace corresponding to the value specified in `.faas.yaml`, or if unset to the currently active Kubernetes namespace. This can be changed with a command line flag. If changed, this value will be persisted in `.faas.yaml`.
```

Comment:
It might be worth noting that a Function can not be deployed unless it is first built (which results in there being an image name in `.faas.yaml`).  To deploy an image built manually, the `.faas.yaml` may be manually edited.